File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
    // Busca el primer ": " y toma el texto después de eso, eliminando espacios en blanco
        return logLine.substring(logLine.indexOf(": ") + 2).trim();
    }

    public static String logLevel(String logLine) {
        int start = logLine.indexOf("[")+1;
        int end = logLine.indexOf("]");
        return logLine.substring(start, end).trim().toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}

File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        return logLine.Split(':',2)[1].Trim();
    }

    public static string LogLevel(string logLine)
    {
        int start = logLine.IndexOf('[')+1;
        int end = logLine.IndexOf(']')-1;
        string level = logLine.Substring(start, end).ToLower();
        return level;
    }

    public static string Reformat(string logLine)
    {
        return Message(logLine) + " (" + LogLevel(logLine) + ")";
    }
}
